### PR TITLE
Add reduce quantity functionality under delete

### LINKED ISSUE #37: As a user, I can reduce the item amounts
... so that I know I can keep track of the amount/quantity of items.

### PR DESCRIPTION
**Pull request by liawsy**
These changes will resolve #37.
Add reduce quantity functionality under delete
Add auto-delete function when quantity of item hits 0
**TODO**
Add test cases
Update user guide

### CODE REVIEW

**Comment by xiaoyu-nus** [comment]
Siew Yee are you showing the correct format message when the quantity to be added or reduced is wrong? I think the correct format should be displayed in the result display window after the error message.

**Comment by liawsy** [comment]
> Siew Yee are you showing the correct format message when the quantity to be added or reduced is wrong? I think the correct format should be displayed in the result display window after the error message.

I have only implemented the reduce quantity function under Delete command. As agreed in the team meeting, we decided not to implement the functionality for quantity to be added, as it is assumed that the user would key in the correct quantity for a particular item of a certain expiry date correctly. Otherwise, the user would need to delete the item and re-add. Do you think I should implement the add quantity functionality under the Add command?

File: src/main/java/io/xpire/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -101,6 +101,10 @@ public CommandResult execute(Model model) throws CommandException, ParseExceptio
             assert this.quantity != null;
             Item newQuantityItem = reduceItemQuantity(targetItem, quantity);
             model.setItem(targetItem, newQuantityItem);
+            /* TODO: Transfer to To-Buy-List*/
+            if (Quantity.quantityIsZero(newQuantityItem.getQuantity())) {
+                model.deleteItem(targetItem);
```
Comment (Kalsyc) [inline src/main/java/io/xpire/logic/commands/DeleteCommand.java]: This isn't a permanent feature right? The item will be moved to the To-Buy List.

Comment (liawsy) [reply src/main/java/io/xpire/logic/commands/DeleteCommand.java]: yep! I've indicated in the comment on line that this is a todo.

**Comment by JermyTan** [comment]
Can use a function in StringUtil to check for >= 0

**Comment by liawsy** [comment]
> To Siew Yee: Sorry I was talking about the add commond message. It would be good if the correct format message for entering stuff can be displayed when the user input is in a wrong format.
> image
> For your delete error message tho, I feel it would be good if you extract it out into a final string.
> And also right now i can add quantity 0. But should only allow positive integer right?

Yep, only positive integers should be allowed as integers. Thank you for pointing this out, I'll make the changes! :) Can you refer me to the line for the particular delete error message that I did not extract into a final string?

**Comment by JermyTan** [comment]
There is also a check for > 0

**Pull request merged**